packages/react-graphql/src/Query.tsx

```diff
@@ -1,13 +1,19 @@
-import React from 'react';
-import {Query as ApolloQuery} from '@apollo/react-components';
 import {OperationVariables} from 'apollo-client';
+import {DocumentNode} from 'graphql-typed';
 
-import {QueryProps} from './types';
+import {useQuery, QueryHookResult, QueryHookOptions} from './hooks';
 
-// eslint-disable-next-line react/prefer-stateless-function
-class QueryTypeClass<
-  Data = any,
-  Variables = OperationVariables
-> extends React.Component<QueryProps<Data, Variables>> {}
+interface QueryComponentOptions<Data, Variables> extends QueryHookOptions {
+  children: (result: QueryHookResult<Data, Variables>) => JSX.Element | null;
+  query: DocumentNode<Data, Variables>;
+}
 
-export const Query: typeof QueryTypeClass = ApolloQuery as any;
+export function Query<Data = any, Variables = OperationVariables>({
+  children,
+  query,
+  ...options
+}: QueryComponentOptions<Data, Variables>) {
+  const result = useQuery<Data, Variables>(query, options);
```

Comment:
I'm having trouble with the query options type:

<img width="869" alt="Screen Shot 2020-06-17 at 6 34 39 PM" src="https://user-images.githubusercontent.com/3865674/84957492-4faeeb00-b0c9-11ea-933f-ec375684824c.png">

@lemonmade wondering if you would mind explaining how [this type](https://github.com/Shopify/quilt/blame/master/packages/react-graphql/src/hooks/query.ts#L29) works?

Reply:
I don't know if it actually does what it says it does, but this type is trying to say that "if all the variable values can be nullish, then this function takes one required argument (the query), and one optional argument (the options, including variables). If at least one variable is required, this function takes two required arguments (the query and the options).

Reply:
Thanks Chris, the explanation makes sense. I'm still having issues passing the query hook options through to the hook. The variables generic type in this case should be the `QueryHookOptions` object since I'm extending it in my `QueryComponentOptions` interface. I'll ask around :)